cheddar/cheddar-integration-aws/src/test/java/com/clicktravel/infrastructure/persistence/aws/dynamodb/DynamoDocumentStoreTemplateTest.java

```diff
@@ -0,0 +1,358 @@
+    @Test
+    public void shouldCreate_withItem() {
+        // Given
+        final ItemId itemId = new ItemId(randomId());
+        final StubItem stubItem = generateRandomStubItem(itemId);
+
+        final ItemConfiguration itemConfiguration = new ItemConfiguration(StubItem.class, tableName);
+        final Collection<ItemConfiguration> itemConfigurations = Arrays.asList(itemConfiguration);
+        when(mockDatabaseSchemaHolder.itemConfigurations()).thenReturn(itemConfigurations);
+
+        final Table mockTable = mock(Table.class);
+        when(mockDynamoDBClient.getTable(any(String.class))).thenReturn(mockTable);
+
+        final DynamoDocumentStoreTemplate dynamoDocumentStoreTemplate = new DynamoDocumentStoreTemplate(
+                mockDatabaseSchemaHolder);
+        dynamoDocumentStoreTemplate.initialize(mockAmazonDynamoDbClient);
+
+        final Item mockTableItem = mock(Item.class);
+        when(mockTableItem.toJSON()).thenReturn(dynamoDocumentStoreTemplate.itemToString(stubItem));
+
+        // When
+        final StubItem returnedItem = dynamoDocumentStoreTemplate.create(stubItem);
+
+        // Then
+        final ArgumentCaptor<PutItemSpec> getItemRequestCaptor = ArgumentCaptor.forClass(PutItemSpec.class);
+        verify(mockTable).putItem(getItemRequestCaptor.capture());
+
+        final PutItemSpec spec = getItemRequestCaptor.getValue();
+        assertEquals(itemId.value(), spec.getItem().get("id"));
+
+        assertEquals(itemId.value(), returnedItem.getId());
+        assertEquals(stubItem.getStringProperty(), returnedItem.getStringProperty());
+        assertEquals(stubItem.getStringProperty2(), returnedItem.getStringProperty2());
+        assertEquals(stubItem.getStringSetProperty(), returnedItem.getStringSetProperty());
+    }
+
+    @SuppressWarnings("deprecation")
+    @Test
+    public void shouldNotCreate_withItem() {
+        // Given
+        final ItemId itemId = new ItemId(randomId());
+        final StubItem stubItem = generateRandomStubItem(itemId);
+
+        final ItemConfiguration itemConfiguration = new ItemConfiguration(StubItem.class, tableName);
+        final Collection<ItemConfiguration> itemConfigurations = Arrays.asList(itemConfiguration);
+        when(mockDatabaseSchemaHolder.itemConfigurations()).thenReturn(itemConfigurations);
+
+        final Table mockTable = mock(Table.class);
+        when(mockDynamoDBClient.getTable(any(String.class))).thenReturn(mockTable);
+
+        final DynamoDocumentStoreTemplate dynamoDocumentStoreTemplate = new DynamoDocumentStoreTemplate(
+                mockDatabaseSchemaHolder);
+        dynamoDocumentStoreTemplate.initialize(mockAmazonDynamoDbClient);
+
+        final Item mockTableItem = mock(Item.class);
+        when(mockTableItem.toJSON()).thenReturn(dynamoDocumentStoreTemplate.itemToString(stubItem));
+
+        doThrow(RuntimeException.class).when(mockTable).putItem(any(PutItemSpec.class));
+
+        exception.expect(RuntimeException.class);
```

Comment:
So far in Cheddar we use a convention for verifying exceptions. Its is slightly more verbose, but we have favoured this approach for clarity:
```java
// Given
...

// When
MyException thrownException = null;
try {
   obj.method(); // method under test
} catch (MyException e) {
   thrownException = e;
}

// Then
assertNotNull(thrownException);
...
```